src/Lecture2.hs

```diff
@@ -70,7 +77,23 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt _ [] = (Nothing, [])
+removeAt n xs 
+  | n < 0 = (Nothing, xs)
+  | otherwise     = (findAtIndex 0 xs, removeAtIndex 0 xs)
```

Comment:
That's a correct solution! 👏🏻 

You can improve it even further by removing the element and returning it during the same single traversal of a list 1️⃣